src/collector/top_collector.rs

```diff
@@ -69,11 +70,21 @@ where
     /// # Panics
     /// The method panics if limit is 0
     pub fn with_limit(limit: usize) -> TopCollector<T> {
```

Comment:
Can we rename `limit` `max_hits`. Limit was not great to begin with... But it is even more confusing now that there is an offset.

Reply:
I'm assuming I should leave the `with_limit` and `limit` functions but just mark them deprecated. Does that sound ok?

Reply:
I've had a bit more of a think about this one I'm unsure about renaming this. I feel like `limit` and `offset` would be more universally recognisable to people and the semantics do match what they would expect. While it is not part of ISO SQL it is the naming used in the most common relational databases (see: [FETCH FIRST SQL clause](https://en.wikipedia.org/wiki/Select_(SQL)#FETCH_FIRST_clause)). Mixing limit and offset wording with something else may cause more confusion with people assuming the semantics of `max_hits` must be different from `limit` because `limit` was not used.

I've done a little investigating in other IR libraries and there is a little less consistency.
- In Lucene their [`TopDocsCollector`](https://lucene.apache.org/core/2_9_4/api/core/org/apache/lucene/search/TopDocsCollector.html) uses `start` and `howMany` (but also `numHits` in the `TopScoreDocCollector`).
- In Elasticsearch they use [`from` and `size`](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/search-request-from-size.html#search-request-from-size).
- In Xapian they use [`pagesize` and `offset`](https://getting-started-with-xapian.readthedocs.io/en/latest/practical_example/searching/building.html) but I think that works better for them because of how their API works.

If you're definitely against using `limit` can I suggest we also use a different word for `offset`? Maybe `start` or `starting_from`.